specs/~dynamic-names.yml

```
@@ -0,0 +1,211 @@
+overview: |
+  Rationale: this special notation was introduced specifically to allow the
+  dynamic loading of partials. The main advantage that this new special notation
+  offers is to allow dynamic loading of partials, which is particularly useful
+  in cases where there needs to be rendered polymorphic data or in cases where
+  there is a child template shared by multiple parent templates; cases which
+  otherwise would be possible to render only by using either complex (and
+  optional) features such as lambdas, or by using solutions that are inefficient
+  both in terms of space and in terms of computational efficiency, such as:
+  overloading the template with if blocks or preprocessing the template.
+
+  Dynamic Names are a special notation to dynamically determine a tag's content.
+
+  Dynamic Names MUST be a non-whitespace character sequence NOT containing
+  the current closing delimiter. A Dynamic Name consists of an asterisk,
+  followed by a dotted name. The latter follows the same notation as in an
+  Interpolation tag.
+
+  This tag's dotted name, which is equal to the Dynamic Name excluding the
+  leading asterisk, refers to a key in the context whose value will be used in
```

```suggestion
  leading asterisk, references a key in the context whose value will be used in
```

(intentionally alluding to c-style references, because that's a model that makes this make sense 🙂)